testdata/config/metrics.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: "config.istio.io/v1alpha2"
+kind: prometheus
+metadata:
+  name: prometheus
+  namespace: istio-config-default
+spec:
+  metrics:
+  - name: requestcount
```

Comment:
pardon my config ignorance: are the names jammed together because certain characters are invalid (or not supported)?

`request_count` is much more readable than `requestcount`. i'd almost prefer `requests` to `requestcount`, but we need a general approach to handling. It also looks a bit... off... to have `requestcount` and `response_code` in the same config stanza.

It seems weird to have the config store dictate the naming used for other backends.

Reply:
k8s is touchy feely about names. The only names that **everyone** likes are all lowercase letters. `request_count` is not allowed, neither is `requestCount`.
All names have to be valid subdomains.  so we **can** use `request-count`, but i don't know where else it will bite us.

Reply:
so, can we add support for aliasing the metrics names (within adapters, I guess) to prometheus and elsewhere?

this is a usability issue, imho.